receiver/awsxrayreceiver/config.go

```diff
@@ -27,24 +27,21 @@ const (
 
 // Config defines the configurations for an AWS X-Ray receiver.
 type Config struct {
-	// The Endpoint field in ReceiverSettings represents the UDP address
+	configmodels.ReceiverSettings `mapstructure:",squash"`
+	// The `NetAddr` represents the UDP address
 	// and port on which this receiver listens for X-Ray segment documents
 	// emitted by the X-Ray SDK.
-	configmodels.ReceiverSettings `mapstructure:",squash"`
-	confignet.TCPAddr             `mapstructure:",squash"`
-
-	// squash ensures fields are correctly decoded in embedded struct
-	// https://godoc.org/github.com/mitchellh/mapstructure#hdr-Embedded_Structs_and_Squashing
+	confignet.NetAddr `mapstructure:",squash"`
```

Comment:
To ensure that it is always UDP I would set it in the default config in the factory, and maybe validate that did not change before creating the receiver

Reply:
Thanks for the review. For now this receiver only has partial implementation, which is why the factory returns an error in `createTraceReceiver` instead of a receiver instance. Your recommended change is in a followup PR that I'm waiting to publish after this one is merged.